### PR TITLE
adds Sigv4 support 

### PR DESCRIPTION
**Pull request by mhausenblas**
This PR addresses #193, adding Sigv4 support for the Prometheus data source.

### CODE REVIEW

File: grafana/resource_data_source_test.go [hunk 1]
```diff
@@ -248,6 +248,9 @@ resource "grafana_data_source" "testdata" {
 			json_data {
 				http_method = "GET"
 				query_timeout = "1"
+                sigv4_auth   = true
+	            sigv4_auth_type = "default"
+	            sigv4_region    = "eu-west-1"
```
Comment (trotttrotttrott) [inline grafana/resource_data_source_test.go]: Whitespace is a bit wonky here.
```suggestion
				sigv4_auth   = true
				sigv4_auth_type = "default"
				sigv4_region    = "eu-west-1"

```

**Review by trotttrotttrott — CHANGES_REQUESTED** [review]
Looks great, just some tidying up to do:

1. Run `go fmt ./grafana/`
2. Run `go generate` - this will update docs/. The content in this directory is generated from schema descriptions and example files. If you'd like to keep the example you added, add it [here](https://github.com/grafana/terraform-provider-grafana/blob/master/examples/resources/grafana_data_source/resource.tf) first. Be sure to also run `terraform fmt` on that file.

**Comment by mhausenblas** [comment]
Thanks a lot for speedy feedback @trotttrotttrott and will do as you advised (not sure why the `go fmt` didn't happen automatically since that's how I've configured it but clearly there is a problem indeed).

**Comment by mhausenblas** [comment]
OK, @trotttrotttrott I think I now did address all your points, PTAL again?

File: docs/resources/data_source.md [hunk 2]
```diff
@@ -43,6 +43,19 @@ resource "grafana_data_source" "cloudwatch" {
...
+  type          = "prometheus"
+  name          = "amp"
+  url           = "https://aps-workspaces.eu-west-1.amazonaws.com/workspaces/ws-1234567890/"
+
+  json_data {
+    http_method     = "POST"
+    sigv4_auth      = true
+    sigv4_auth_type = "default"
+    sigv4_region    = "eu-west-1"
+   }
```
Comment (trotttrotttrott) [inline docs/resources/data_source.md]: `terraform fmt`...

```suggestion
  type = "prometheus"
  name = "amp"
  url  = "https://aps-workspaces.eu-west-1.amazonaws.com/workspaces/ws-1234567890/"

  json_data {
    http_method     = "POST"
    sigv4_auth      = true
    sigv4_auth_type = "default"
    sigv4_region    = "eu-west-1"
  }
```

File: examples/resources/grafana_data_source/resource.tf [hunk 3]
```diff
@@ -22,6 +22,19 @@ resource "grafana_data_source" "cloudwatch" {
...
+  type          = "prometheus"
+  name          = "amp"
+  url           = "https://aps-workspaces.eu-west-1.amazonaws.com/workspaces/ws-1234567890/"
+
+  json_data {
+    http_method     = "POST"
+    sigv4_auth      = true
+    sigv4_auth_type = "default"
+    sigv4_region    = "eu-west-1"
+   }
```
Comment (trotttrotttrott) [inline examples/resources/grafana_data_source/resource.tf]:
```suggestion
  type = "prometheus"
  name = "amp"
  url  = "https://aps-workspaces.eu-west-1.amazonaws.com/workspaces/ws-1234567890/"

  json_data {
    http_method     = "POST"
    sigv4_auth      = true
    sigv4_auth_type = "default"
    sigv4_region    = "eu-west-1"
  }
```

File: docs/resources/data_source.md [hunk 4]
```diff
@@ -131,8 +150,9 @@ Optional:
 - **password** (String, Sensitive) (All) Password to use for authentication.
 - **private_key** (String, Sensitive) (Stackdriver) The service account key `private_key` to use to access the data source.
 - **secret_key** (String, Sensitive) (CloudWatch) The secret key to use to access the data source.
+- **sigv4_access_key** (String, Sensitive) (Elasticsearch and Prometheus) SigV4 access key. Required when using 'keys' auth provider.
+- **sigv4_secret_key** (String, Sensitive) (Elasticsearch and Prometheus) SigV4 secret key. Required when using 'keys' auth provider.
 - **tls_ca_cert** (String, Sensitive) (All) CA cert for out going requests.
 - **tls_client_cert** (String, Sensitive) (All) TLS Client cert for outgoing requests.
 - **tls_client_key** (String, Sensitive) (All) TLS Client key for outgoing requests.
```
Comment (trotttrotttrott) [inline docs/resources/data_source.md:158]: `tfplugindocs` wants this space here. Adding back so `go generate` doesn't produce a diff.

```suggestion


```

**Review by trotttrotttrott — APPROVED** [review]

**Pull request merged**